barretenberg/cpp/src/barretenberg/protogalaxy/protogalaxy_prover.hpp

```diff
@@ -6,62 +6,46 @@ namespace bb {
-            typename Flavor::template OptimisedProtogalaxyTupleOfTuplesOfUnivariates<NUM_INSTANCES>;
-        using RelationParameters = bb::RelationParameters<Univariate<FF, ProverInstances_::EXTENDED_LENGTH>>;
-        using OptimisedRelationParameters = bb::RelationParameters<
-            Univariate<FF, ProverInstances_::EXTENDED_LENGTH, 0, /*skip_count=*/NUM_INSTANCES - 1>>;
-        using CombinedRelationSeparator =
-            std::array<Univariate<FF, ProverInstances_::BATCHED_EXTENDED_LENGTH>, Flavor::NUM_SUBRELATIONS - 1>;
+    using ProverInstance = typename ProverInstances_::Instance;
+    using Flavor = typename ProverInstances_::Flavor;
+    using FF = typename ProverInstances_::Flavor::FF;
+    static constexpr size_t NUM_INSTANCES = ProverInstances_::NUM;
+    using CombinerQuotient = Univariate<FF, ProverInstances_::BATCHED_EXTENDED_LENGTH, NUM_INSTANCES>;
+    using TupleOfTuplesOfUnivariates = typename Flavor::template ProtogalaxyTupleOfTuplesOfUnivariates<NUM_INSTANCES>;
+    using OptimisedTupleOfTuplesOfUnivariates =
+        typename Flavor::template OptimisedProtogalaxyTupleOfTuplesOfUnivariates<NUM_INSTANCES>;
+    using UnivariateRelationParameters =
+        bb::RelationParameters<Univariate<FF, ProverInstances_::EXTENDED_LENGTH, 0, /*skip_count=*/NUM_INSTANCES - 1>>;
+    using UnivariateRelationSeparator =
+        std::array<Univariate<FF, ProverInstances_::BATCHED_EXTENDED_LENGTH>, Flavor::NUM_SUBRELATIONS - 1>;
 
+    struct State {
         std::shared_ptr<ProverInstance> accumulator;
         LegacyPolynomial<FF> perturbator;
-        std::vector<FF> gate_challenges;
         std::vector<FF> deltas;
         CombinerQuotient combiner_quotient;
-        FF compressed_perturbator;
-        RelationParameters relation_parameters;
-        CombinedRelationSeparator alphas; // a univariate interpolation of challenges for each subrelation
-        OptimisedRelationParameters optimised_relation_parameters;
-        OptimisedTupleOfTuplesOfUnivariates optimised_univariate_accumulators;
-        TupleOfTuplesOfUnivariates univariate_accumulators;
-        FoldingResult<typename ProverInstances_::Flavor> result;
+        FF perturbator_evaluation;
+        UnivariateRelationParameters relation_parameters;
+        UnivariateRelationSeparator alphas;
     };
-
-    using ProverInstances = ProverInstances_;
-    using Flavor = typename ProverInstances::Flavor;
     using Transcript = typename Flavor::Transcript;
-    using FF = typename Flavor::FF;
-    using Instance = typename ProverInstances::Instance;
+    using Instance = typename ProverInstances_::Instance;
     using CommitmentKey = typename Flavor::CommitmentKey;
+    using ProverInstances = ProverInstances_;
 
-    static constexpr size_t NUM_SUBRELATIONS = ProverInstances::NUM_SUBRELATIONS;
+    static constexpr size_t NUM_SUBRELATIONS = ProverInstances_::NUM_SUBRELATIONS;
 
-    ProverInstances instances;
+    ProverInstances_ instances;
```

Comment:
_ not needed?

Reply:
I was running into trouble where the compiler was complaining about two functions having different return types when the type was actually the same but we were accessing it through different aliases. This was part of the fix. I have a plan to deal with this, namely I'm going to make the template take more types, and then proxy through another template as in 
```
template <typename Flavor, typename FF> C_;
using template <typename Flavor> C = C_<Flavor, typename Flavor::FF>;
```
This will make a lot of the complicated signatures much more readable.